# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/PQuota.ipdl
# Commit: 46a5babaafd2
# Full Hash: 46a5babaafd2c0f2198f353d62254cf518dab391
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-15 04:16:11
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Implement initialization of entire persistent storage; r=dom-storage-reviewers,jari
#   
#   IndexedDB still calls QuotaManager::EnsurePersistentOriginIsInitializedInternal
#   from its idle database maintenance. However, we want to get rid of that so the
#   method can be made private. The idle database maintenance should only rely on
# ==============================================================================

diff -r 2ff5f3aaf841 -r 46a5babaafd2 dom/quota/PQuota.ipdl
--- a/dom/quota/PQuota.ipdl	Mon Oct 14 10:00:26 2024 +0000
+++ b/dom/quota/PQuota.ipdl	Mon Oct 14 10:06:14 2024 +0000
@@ -125,6 +125,9 @@
   async StorageInitialized()
       returns(BoolResponse response);
 
+  async PersistentStorageInitialized()
+      returns(BoolResponse response);
+
   async TemporaryStorageInitialized()
       returns(BoolResponse response);
 
@@ -138,6 +141,9 @@
   async InitializeStorage()
       returns(BoolResponse response);
 
+  async InitializePersistentStorage()
+      returns(BoolResponse response);
+
   async InitializePersistentOrigin(PrincipalInfo principalInfo)
       returns(BoolResponse response);
 